instrumentation/net/http/otelhttp/handler.go

```diff
@@ -35,18 +35,19 @@ type middleware struct {
 	operation string
 	server    string
 
-	tracer            trace.Tracer
-	meter             metric.Meter
-	propagators       propagation.TextMapPropagator
-	spanStartOptions  []trace.SpanStartOption
-	readEvent         bool
-	writeEvent        bool
-	filters           []Filter
-	spanNameFormatter func(string, *http.Request) string
-	counters          map[string]metric.Int64Counter
-	valueRecorders    map[string]metric.Float64Histogram
-	publicEndpoint    bool
-	publicEndpointFn  func(*http.Request) bool
+	tracer               trace.Tracer
+	meter                metric.Meter
+	propagators          propagation.TextMapPropagator
+	spanStartOptions     []trace.SpanStartOption
+	readEvent            bool
+	writeEvent           bool
+	filters              []Filter
+	spanNameFormatter    func(string, *http.Request) string
+	requestBytesCounter  metric.Int64Counter
+	responseBytesCounter metric.Int64Counter
+	serverLatencyMeasure metric.Float64Histogram
+	publicEndpoint       bool
+	publicEndpointFn     func(*http.Request) bool
```

Comment:
nit: I propose to make a separate "section" for the instruments


```suggestion
	tracer            trace.Tracer
	meter             metric.Meter
	propagators       propagation.TextMapPropagator
	spanStartOptions  []trace.SpanStartOption
	readEvent         bool
	writeEvent        bool
	filters           []Filter
	spanNameFormatter func(string, *http.Request) string
	publicEndpoint    bool
	publicEndpointFn  func(*http.Request) bool
	
	requestBytesCounter  metric.Int64Counter
	responseBytesCounter metric.Int64Counter
	serverLatencyMeasure metric.Float64Histogram
```